opm/grid/LookUpData.hh

```diff
@@ -277,6 +308,42 @@ Opm::LookUpData<Grid,GridView>::operator()(const EntityType& elem,
     return fieldProp[fieldPropIdx];
 }
 
+template<typename Grid, typename GridView>
+std::vector<double> Opm::LookUpData<Grid,GridView>::assignFieldPropsDoubleOnLeaf(const FieldPropsManager& fieldPropsManager,
+                                                                                 const std::string& propString,
+                                                                                 const unsigned int& numElements,
+                                                                                 const double& defaultValue) const
+{
+    std::vector<double> fieldPropOnLeaf;
+    fieldPropOnLeaf.resize(numElements, defaultValue);
+    const auto& fieldProp = fieldPropsManager.get_double(propString);
+    for (unsigned int elemIdx = 0; elemIdx < numElements; ++elemIdx) {
+        const auto& fieldPropIdx = this->getFieldPropIdx<Grid>(elemIdx);
+        fieldPropOnLeaf[elemIdx] = fieldProp[fieldPropIdx];
+    }
+    return fieldPropOnLeaf;
+}
+
+template<typename Grid, typename GridView>
+template<typename IntType>
+std::vector<IntType> Opm::LookUpData<Grid,GridView>::assignFieldPropsIntOnLeaf(const FieldPropsManager& fieldPropsManager,
+                                                                               const std::string& propString,
+                                                                               const unsigned int& numElements,
+                                                                               const int& defaultValue,
+                                                                               const bool& needsTranslation,
+                                                                               std::function<void(IntType)> valueCheck) const
+{
+    std::vector<IntType> fieldPropOnLeaf;
+    fieldPropOnLeaf.resize(numElements, defaultValue);
+    const auto& fieldProp = fieldPropsManager.get_int(propString);
+    for (unsigned elemIdx = 0; elemIdx < numElements; ++elemIdx) {
+        const auto& fieldPropIdx = this->getFieldPropIdx<Grid>(elemIdx);
+        fieldPropOnLeaf[elemIdx] = fieldProp[fieldPropIdx] - needsTranslation;
+        valueCheck(fieldPropOnLeaf[elemIdx]);
```

Comment:
Should we do the check with the original value?

Reply:
It needs to be more general for sure. For the value-check of rockTableIdx_ in eclgenericproblem_impl.hh, I also need to take into account the course-element-index. So there is more to do here. 

I keep in mind the possibility of doing the check with the non-translated value.